cpp/include/cudf/io/text/detail/tile_state.hpp

```diff
@@ -0,0 +1,174 @@
+
+#pragma once
+
+#include <cub/block/block_scan.cuh>
+
+namespace cudf {
+namespace io {
+namespace text {
+namespace detail {
+
+enum class scan_tile_status : uint8_t {
+  oob,
+  invalid,
+  partial,
+  inclusive,
+};
+
+template <typename T>
+struct scan_tile_state_view {
+  uint64_t num_tiles;
+  scan_tile_status* tile_status;
+  T* tile_partial;
+  T* tile_inclusive;
+
+  __device__ inline void initialize_status(cudf::size_type base_tile_idx,
+                                           cudf::size_type count,
+                                           scan_tile_status status)
+  {
+    auto thread_idx = blockIdx.x * blockDim.x + threadIdx.x;
+    if (thread_idx < count) {  //
+      // this is UB if tile_status gets assigned from multiple threads.
```

Comment:
Is it UB? Given that threads would be concurrently writing *the same* value, I would have expect this to be ok?

Reply:
@elstehle It is technically UB for multiple threads to write to the same memory location regardless of whether they're storing the same value. This is an often debated topic, but the bottom line is _don't do it_, because the memory model makes no guarantees as to the outcome being what you expect it would be.

@devavret I think if launch with an appropriately sized `count`, we should be fine. The comment here is just to remind us that we have to do that. I'll update the comment to make that restriction more clear.